source/dofs/dof_handler_policy.cc

```diff
@@ -1599,10 +1599,10 @@ namespace internal
                                    const std::vector<dealii::types::global_dof_index> &p4est_tree_to_coarse_cell_permutation)
         {
           // build list of cells to request for each neighbor
-          std::set<unsigned int> level_ghost_owners = tria.level_ghost_owners();
+          std::set<dealii::types::subdomain_id> level_ghost_owners = tria.level_ghost_owners();
```

Comment:
This is in an `internal` namespace that happens to have its own sub-namespace `types`. If I don't qualify with `dealii::`, then `types` refers to the local namespace in which there is no `subdomain_id`. I had to figure this out as well, but there are a number of places in the vicinity that do the same -- for example in line 1599, just above.